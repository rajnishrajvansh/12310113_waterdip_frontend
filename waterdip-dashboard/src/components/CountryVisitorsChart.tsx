
import React from 'react';
import Chart from 'react-apexcharts';
import { ApexOptions } from 'apexcharts';

interface CountryVisitorsChartProps {
  data: { country: string; visitors: number }[];
}

const CountryVisitorsChart = ({ data }: CountryVisitorsChartProps) => {
  const options: ApexOptions = {
    chart: { type: 'bar', height: 350, },
    plotOptions: {
      bar: {
        borderRadius: 5,
        dataLabels: {
          position: 'top', // top, center, bottom
        },
      }
    },
    dataLabels: {
      enabled: true,
      offsetY: -20,
      style: {
        fontSize: '12px',
        colors: ["#304758"]
      }
    },
    xaxis: {
       categories: data.map(item => item.country), 
       position:'top',
       axisBorder: {
        show: false
      },
      axisTicks: {
        show: false
      },
      crosshairs: {
        fill: {
          type: 'gradient',
          gradient: {
            colorFrom: '#D8E3F0',
            colorTo: '#BED1E6',
            stops: [0, 100],
            opacityFrom: 0.4,
            opacityTo: 0.5,
          }
        }
      },
      tooltip: {
        enabled: true,
      }
      },
      yaxis: {
        title: {
          text: 'Visitors'
        },
        axisBorder: {
          show: true
        },
        axisTicks: {
          show: false,
        },
        labels: {
          show: false,
        }
      
      },
      title: {
        text: 'Country Visitors Chart',
        floating: true,
        offsetY: 330,
        align: 'center',
        style: {
          color: '#444'
        }
      }
  };

  const series = [
    {
      name: 'Visitors',
      data: data.map(item => item.visitors),
    },
  ];

  return <Chart options={options} series={series} type="bar" />;
};

export default CountryVisitorsChart;
