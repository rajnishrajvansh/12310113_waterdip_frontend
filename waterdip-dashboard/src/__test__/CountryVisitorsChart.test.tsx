
import React from 'react';
import { render } from '@testing-library/react';
import CountryVisitorsChart from '../components/CountryVisitorsChart';

const countryData = [
  { country: 'USA', visitors: 100 },
  { country: 'UK', visitors: 50 },
];

test('matches snapshot for CountryVisitorsChart', () => {
  const { asFragment } = render(<CountryVisitorsChart data={countryData} />);
 
  expect(asFragment()).toMatchSnapshot();
});
