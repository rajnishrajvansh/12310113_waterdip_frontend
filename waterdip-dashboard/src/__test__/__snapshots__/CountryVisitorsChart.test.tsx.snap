// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for CountryVisitorsChart 1`] = `
<DocumentFragment>
  <div
    height="auto"
    options="[object Object]"
    series="[object Object]"
    type="bar"
    width="100%"
  />
</DocumentFragment>
`;
